tutorial/consuming_packages/getting_started/conanfile_py/conanfile.py

```diff
@@ -1,17 +1,35 @@
+import os
+
 from conan import ConanFile
 from conan.tools.cmake import cmake_layout
+from conan.errors import ConanInvalidConfiguration
 
-
-class compressorRecipe(ConanFile):
+class CompressorRecipe(ConanFile):
     # Binary configuration
     settings = "os", "compiler", "build_type", "arch"
     generators = "CMakeToolchain", "CMakeDeps"
 
+    def validate(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("ARM v8 not supported")
+
     def requirements(self):
         self.requires("zlib/1.2.11")
+        # Use the system's CMake for Windows
+        # and add base64 dependency
+        if self.settings.os == "Windows":
+            self.requires("base64/0.4.0")
+        else:
+            self.tool_requires("cmake/3.19.8")
 
-    # do not introduce layout method yet
-    # use conan install . --output-folder cmake-build-release
-    # then explain you can use layout to pre-define that
-    # def layout(self):
-    #     cmake_layout(self)
+    def layout(self):
+        build_type = str(self.settings.build_type)
```

Comment:
Can't use a built-in layout? Why a custom one is needed?

Reply:
Because the docs explain how to transition from output-folder to layouts and to introduce the users to the layout feature in a simple way